scripts/aws_api_key_manager.py

```diff
@@ -0,0 +1,335 @@
+            UserName=user_name)
+
+        all_keys = existing_keys['AccessKeyMetadata']
+        keys_list = []
+
+        if all_keys:
+            for ekey in all_keys:
+                keys_list.append(ekey['AccessKeyId'])
+
+        return keys_list
+
+
+    @staticmethod
+    def get_token(aws_account):
+        """ generate temporary SSO access credentials  """
+
+        sso_config_path = '/etc/openshift_tools/sso-config.yaml'
+
+        if os.path.isfile(sso_config_path):
+            with open(sso_config_path, 'r') as sso_config:
+                yaml_config = yaml.load(sso_config)
+
+                if yaml_config["idp_host"]:
+                    ops_idp_host = yaml_config["idp_host"]
+
+                creds = saml_aws_creds.get_temp_credentials(
+                    metadata_id='urn:amazon:webservices:%s' % aws_account,
+                    idp_host=ops_idp_host
+                    )
+
+                client = boto3.client(
+                    'iam',
+                    aws_access_key_id=creds['AccessKeyId'],
+                    aws_secret_access_key=creds['SecretAccessKey'],
+                    aws_session_token=creds['SessionToken']
+                    )
+
+                return client
+
+        else:
+            raise ValueError(sso_config_path + 'does not exist')
+
+    @staticmethod
+    def create_key(aws_account, user_name, client):
+        """ change an API key for the specified account"""
+
+        response = client.create_access_key(
+            UserName=user_name
+            )
+
+        print('key successfully created for:', aws_account)
+        return response
+
+
+    @staticmethod
+    def delete_key(aws_account, user_name, key, client):
+        """ delete an API key for the specified account"""
+
+        response = client.delete_access_key(
+            UserName=user_name,
+            AccessKeyId=key
+            )
+
+        print('key successfully deleted for:', aws_account)
+        return response
+
+
+    @staticmethod
+    def manage_timestamp(update=False):
+        """ create or update expiration file """
+
+        path = os.path.join(os.path.expanduser('~'), '.aws/credentials_expiration')
+        exp_date = str(int(time.time())+180*24*60*60)
+
+        if os.path.isfile(path) and update is True:
+            print('file exists, ready to write')
+            with open(path, 'w') as open_file:
+                open_file.write(exp_date)
+
+        elif not os.path.isfile(path):
+            print('file does not exist, creating')
+            with open(path, 'w') as open_file:
+                open_file.write(exp_date)
+
+        else:
+            print('checked for file and it exists. no write was called. nothing to do here')
+            return True
+
+
+    @staticmethod
+    def write_credentials(aws_account, key_object):
+        ''' write the profile for the user account to the AWS credentials file '''
+
+        path = os.path.join(os.path.expanduser('~'), '.aws/credentials')
```

Comment:
I suggest using os.environ('HOME') instead of expanduser('~'). It's clearer what the intent is.